storage/innobase/trx/trx0purge.cc

```diff
@@ -497,6 +497,7 @@ inline dberr_t purge_sys_t::iterator::free_history_rseg(trx_rseg_t &rseg) const
       if (UNIV_UNLIKELY(mach_read_from_4(TRX_RSEG + TRX_RSEG_FORMAT +
                                          rseg_hdr->page.frame)))
         trx_rseg_format_upgrade(rseg_hdr, &mtr);
+      DBUG_EXECUTE_IF("skip_cached_undo", goto skip_purge_free;);
```

Comment:
This indeed corresponds to reverting 86767bcc0f121db3ad83a74647a642754a0ce57f. Well done.